tests/nested/manual/core20-da-lockout/task.yaml

```diff
@@ -0,0 +1,39 @@
+summary: Check that an unclean shutdown on UC20 devices works
+
+details: |
+    This test checks that an unclean shutdown of a UC20 device does
+    not cause an eventual DA lockout (LP: 1979185)
+
+systems: [ubuntu-20.04-64, ubuntu-22.04-64]
+
+# This test takes about 2h to run so we won't enable it by default
+manual: true
+
+kill-timeout: 3h
+
+environment:
+  NESTED_ENABLE_TPM/encrypted: "true"
+  NESTED_ENABLE_SECURE_BOOT/encrypted: "true"
+
+execute: |
+    tests.nested build-image core
+    tests.nested create-vm core
+
+    tests.nested exec "sudo snap wait system seed.loaded"
+    tests.nested wait-for device-initialized
+
+    # snapcore/secboot has code in tpm2/provisioning.go that sets
+    # maxTries=32 for the DA lockout.  So when uncleanly shutdown for
+    # 33 times we trigger the da lockout (see
```

Comment:
We can probably figure out a way to reduce this for the test, as it can be changed as long as you have access to the lockout hierarchy authorization value. Is it possible to have additional snaps inside the VM to give us the tools to do this?

Reply:
Yeah, we can run arbitrary code in this test, I could build my own go program for example as part of the test

Reply:
The other possibility here is that rather than running the test 33 times to make sure the DA lockout isn't triggered, you could run it fewer times and read the lockout counter value back from the TPM to make sure that it's equal to 0.